File: python/7_kyu/simple_string_reversal_II.py
# Simple string reversal II
# https://www.codewars.com/kata/5a8d1c82373c2e099d0000ac/train/python

def solve(st,a,b):
    first = []
    last = []
    rev = []
    
    for i in range(0, len(st)):
        if i < a:
            first.append(st[i])
        elif a <= i <= b:
            rev.append(st[i])
        else:
            last.append(st[i])
    
    rev.reverse()
    
    x = ''.join(first)
    y = ''.join(rev)
    z = ''.join(last)
    
    return x + y + z
            

File: javascript/7_kyu/simple_string_reversal_II.js
// Simple string reversal II
// https://www.codewars.com/kata/5a8d1c82373c2e099d0000ac/train/javascript

function solve(st,a,b){
    var first = [];
    var last = [];
    var flip = [];
    
    for (var i = 0; i < st.length; i++) {
      if (i >= a && i <= b) {
        flip.push(st[i]);
      } else if (i < a) {
        first.push(st[i]);
      } else {
        last.push(st[i]);
      }
    }
    
    flip.reverse();
    
    var x = first.join('');
    var y = flip.join('');
    var z = last.join('');
    
    return x + y + z;
  }
    